connector_amazon/models/amazon_sale_importer.py

```diff
@@ -3,14 +3,204 @@
 # @author Sébastien BEAU <sebastien.beau@akretion.com>
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
+import StringIO
+import logging
 
-from openerp import models
+from openerp import _, models
+from openerp.exceptions import Warning as UserError
+
+_logger = logging.getLogger(__name__)
+
+try:
+    import unicodecsv
+except (ImportError, IOError) as err:
+    _logger.debug(err)
 
 
 class AmazonSaleImporter(models.AbstractModel):
     _name = 'amazon.sale.importer'
     _description = 'Amazon Sale Importer'
 
-    def _run(self, report):
-        # TODO process the report and generate the sale order
-        pass
+    def _run(self, report, meta_attachment):
+        """ Process the report and generate the sale order
+        """
+        fieldnames = [
+            'order-id', 'order-item-id', 'purchase-date', 'payments-date',
+            'buyer-email', 'buyer-name', 'buyer-phone-number', 'sku',
+            'product-name', 'quantity-purchased', 'currency', 'item-price',
+            'item-tax', 'shipping-price', 'shipping-tax',
+            'ship-service-level', 'recipient-name', 'ship-address-1',
+            'ship-address-2', 'ship-address-3', 'ship-city', 'ship-state',
+            'ship-postal-code', 'ship-country', 'ship-phone-number',
+            'delivery-start-date', 'delivery-end-date',
+            'delivery-time-zone', 'delivery-Instructions', 'sales-channel',
+        ]
+        file = StringIO.StringIO()
+        file.write(report)
+        file.seek(0)
+        reader = unicodecsv.DictReader(
+            file, fieldnames=fieldnames, delimiter='\t', quoting=False,
+            encoding=meta_attachment.amazon_backend_id.encoding)
+        reader.next()  # we pass the file header
+        sales = self._extract_infos(reader)
+        file.close()
+        return self._create_sales(sales, meta_attachment)
+
+    def _extract_infos(self, reader):
+        sales = {}
+
+        def reset_empty(value):
+            if value == '':
+                return False
+            return value
+
+        for line in reader:
+            if not line.get('order-item-id'):
+                continue
+            if line['order-id'] in sales:
+                vals = sales[line['order-id']]
+                vals['lines'].append(self._get_sale_line(line))
```

Comment:
1/ `sales[line['order-id']]['lines'].append(...)` vs `vals['lines'].append(...)` it's a question of taste, no ? You're a https://en.wikipedia.org/wiki/Captain_Hook fan, Seb ?
My eyes are stitching with too many `[` for me and other old people ;-)

2/ cost yes it's done below